gosnmp.go

```diff
@@ -119,6 +119,10 @@ type GoSNMP struct {
 	// we open unconnected UDP socket and use sendto/recvfrom.
 	UseUnconnectedUDPSocket bool
 
+	// Used if only UseUnconnectedUDPSocket if set
+	// Default ":0" - bind to 0.0.0.0 with random port
+	Listen string
```

Comment:
Let's call this `ListenAddress` to match the naming used by the `net` package.

Reply:
Here I thought that by the way it is possible to implement a local ip:port binding not only for a UDP socket of the listening type (with UseUnconnectedUDPSocket), but also for ordinary sockets, both tcp and udp, by setting a variable in net.Dialer {Timeout: x .Timeout, LocalAddr: <addr>}.
Then maybe it makes sense to name the variable, respectively, LocalAddress or LocalListenAddress? Although I have no objections to ListenAddress.

Reply:
I agree:
- when `UseUnconnectedUDPSocket` is false, this option should set `LocalAddr` in the [net.Dialer](https://pkg.go.dev/net#Dialer) struct, which makes it work for normal "connected" UDP sockets, and for TCP connections too
- I think it makes sense to name the GoSNMP option `LocalAddr` too. After all, it's not "listening" for incoming SNMP queries - this option is used to set the source address for *outbound* SNMP queries.

My use case for this feature is described at https://github.com/prometheus/snmp_exporter/issues/683. I have a loopback interface and I want to send SNMP queries with the loopback address as the source.  The target devices are well-behaved so I don't have to worry about the replies coming back with the wrong source address (which is is what I understand `UseUnconnectedUDPSocket` is for)